src/main/kotlin/org/vitrivr/cottontail/math/knn/metrics/Keypoints2DDistance.kt

```diff
@@ -0,0 +1,56 @@
+package org.vitrivr.cottontail.math.knn.metrics
+
+import org.vitrivr.cottontail.model.values.DoubleValue
+import org.vitrivr.cottontail.model.values.types.RealVectorValue
+import org.vitrivr.cottontail.model.values.types.VectorValue
+
+/**
+ * Calculates the Keypoints 2D distance between two vectors.
+ * This is the sum of Euclidian distances between consecutive pairs of points in a vector
+ *
+ * @version 1.0
+ * @author Ralph Gasser
+ */
+object Keypoints2DDistance : DistanceKernel {
+    override val cost = 1.0f
+
+    private fun procDiffs(diffs: RealVectorValue<*>): DoubleValue {
+        val sqDiffs = diffs.pow(2)
+        sqDiffs.indices.endInclusive - sqDiffs.indices.start + 1
```

Comment:
Does this statement have any effect?

Reply:
Whoops. Looks like I got a bit mixed up with the previous PR.